webextension/background.js

```diff
@@ -333,30 +347,62 @@ const themeManager = {
 };
 
 const tabPageCounter = {
-  counter: {},
-
-  initTabCounter(tab) {
-    if (tab.id in this.counter) {
+  counters: {},
+
+  initTabCounter(tab, why = "user-activated-tab") {
+    // When the user becomes active,
+    // we ONLY need to initialize the activity counter
+    if (tab.id in this.counters && why === "user-became-active") {
+      this.counters[tab.id].activity = {
+        "cookieStoreId": tab.cookieStoreId,
+        "pageRequests": 0
+      };
+      return;
+    }
+    if (tab.id in this.counters) {
       return;
     }
-    this.counter[tab.id] = {
+    this.counters[tab.id] = {};
+    this.counters[tab.id].tab = {
+      "cookieStoreId": tab.cookieStoreId,
+      "pageRequests": 0
+    };
+    this.counters[tab.id].activity = {
       "cookieStoreId": tab.cookieStoreId,
       "pageRequests": 0
     };
   },
 
-  sendTabCountAndDelete(tabId) {
-    browser.runtime.sendMessage({
-      method: "sendTelemetryPayload",
-      event: "page-requests-completed-per-tab",
-      userContextId: this.counter[tabId].cookieStoreId,
-      pageRequestCount: this.counter[tabId].pageRequests
-    });
-    delete this.counter[tabId];
+  sendTabCountAndDelete(tabId, why = "user-closed-tab") {
+    if (why === "user-closed-tab") {
+      browser.runtime.sendMessage({
+        method: "sendTelemetryPayload",
+        event: "page-requests-completed-per-tab",
+        userContextId: this.counters[tabId].tab.cookieStoreId,
+        pageRequestCount: this.counters[tabId].tab.pageRequests
```

Comment:
It's getting complicated, so I filed https://github.com/mozilla/testpilot-containers/issues/446 for this.